syne_tune/optimizer/schedulers/searchers/searcher.py

```diff
@@ -595,15 +658,43 @@ def _generate_remaining_candidates(self) -> List[dict]:
         return remaining_candidates
 
     def _generate_all_candidates_on_grid(self) -> List[dict]:
-        # Get hp values that are specified as Domain
+        """
+        Generates all configurations to be evaluated by placing a regular, equally spaced grid over the configuration space
+        Returns:
+            The set of all configurations on grid.
+        """
         hp_keys = []
         hp_values = []
-        for key, hp_range in reversed(list(self.config_space.items())):
-            if isinstance(hp_range, Domain):
-                hp_keys.append(key)
+        ## adding categorical, finiteRange, scalar parameters
+        for hp, hp_range in reversed(list(self.config_space.items())):
+            if isinstance(hp_range, Float) or isinstance(hp_range, Integer):
+                continue
+            if isinstance(hp_range, Categorical):
+                hp_keys.append(hp)
```

Comment:
Move this 2 lines up, removes duplicatons

Reply:
I changed it as this:
            if isinstance(hp_range, Categorical):
                values = hp_range.categories
            elif isinstance(hp_range, FiniteRange):
                values = hp_range.values
            elif not isinstance(hp_range, Domain):
                values = [hp_range]
            hp_keys.append(hp)
            hp_values.append(values)

but i wonder could we have a case that we need to catch in an else statement?

Reply:
Ah I see, then maybe your old code is OK